presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/ExtendedHiveMetastore.java

```diff
@@ -133,12 +132,12 @@ List<PartitionNameWithVersion> getPartitionNamesWithVersionByFilter(
 
     default long lock(MetastoreContext metastoreContext, String databaseName, String tableName)
     {
-        throw new NotSupportedException("Lock is not supported by default");
+        return 1L;
```

Comment:
I think we can keep the default as it is in this interface and then handle lock & unlock in Glue implementation respectively instead.

And for https://github.com/prestodb/presto/blob/master/presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/ExtendedHiveMetastore.java#L134
We can make this Optional and return empty for Glue.
Let me know what do you think?

Reply:
Yes, I agree. Let's keep the interface as-is since we don't want to allow pass-through by default in case of other metastores. We can override this in Glue Hive Metastore and allow pass-through.

Also, let's add some details about the verifications that you did to showcase we can pass through in the case of Glue Metastore